src/integrator/LangevinThermostat.cpp

```diff
@@ -128,11 +129,16 @@ namespace espressopp {
       LOG4ESPP_DEBUG(theLogger, "thermalize");
 
       System& system = getSystemRef();
-      
+
       CellList cells = system.storage->getRealCells();
 
       for(CellListIterator cit(cells); !cit.isDone(); ++cit) {
-        frictionThermo(*cit);
+
+        if(exclusions.count((*cit).id()) == 0)
```

Comment:
We're adding extra overhead here for a situation that rarely occurs (I guess 99% of the time people will use thermostat with no exclusions), but I guess if length(exclusions)==0 then the call to count has almost no cost...